packages/core/src/notifications/panelNode.ts

```diff
@@ -187,49 +187,70 @@ export class NotificationsNode implements TreeNode {
         }
     }
 
-    public async renderModal(notification: ToolkitNotification) {
-        getLogger('notifications').info('rendering modal ... ')
-        if (!notification.uiRenderInstructions.buttons) {
+    /**
+     * Renders information window with the notification's content and buttons.
+     * Can be either a blocking modal or a bottom-right corner toast
+     * Handles the button click actions based on the button type.
+     */
+    public async showInformationWindow(notification: ToolkitNotification, type: string = 'toast') {
+        const isModal = type === 'modal'
+
+        // modal has to have defined actions(buttons)
+        if (!notification.uiRenderInstructions.actions && isModal) {
```

Comment:
Why do we need to enforce this? A modal with no actions can just have a simple 'OK' button that is built in by VSC.

Reply:
In that case yea I guess we don't have to enforce this.